apps/comps/app/api/login/route.ts

```diff
@@ -0,0 +1,48 @@
+import {NextRequest, NextResponse} from 'next/server';
+import {SiweMessage} from 'siwe';
+import {verifyMessage} from '@wagmi/core';
+import {config} from "@/wagmi-config";
+
+const TIME_LIMIT = 24 * 3600 * 1000 // 1 day before sig expires
+
+const validateMessage = (req: NextRequest, msg: SiweMessage) => {
+  const host = req.headers.get("host"); // e.g. "localhost:3000"
+
+  if (msg.domain != host) {
+    throw new Error("Invalid domain")
+  }
+  if (msg.chainId != 84532) {
+    throw new Error("Invalid chain id")
+  }
+  if (msg.nonce != '' && false) { // we should get nonce from some db
+    throw new Error("Invalid nonce")
+  }
```

Comment:
about the nonce, what we can do for now is something similar to https://github.com/spruceid/siwe-quickstart/blob/main/03_complete_app/backend/src/index.js which saves the generated nonce in a cookie and then we verify the signed message against that nonce (https://viem.sh/docs/siwe/actions/verifySiweMessage#nonce-optional)

Reply:
So just to clarify, in the api we will need a `nonce` endpoint that is called by the client before constructing the SIWE payload on the client side. The endpoint stores the nonce encrypted in a cookie and also returns the nonce to the client. The client uses that nonce in the payload, then calls the `login` endpoint. The `login` endpoint decrypts the cookie to get the nonce, and verifies the SIWE payload with that nonce.

Reply:
Ah cool, I see the nonce enddpoint below now, cool.